tap-agent/src/agent/allocation_id_tracker.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023-, GraphOps and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use alloy_primitives::Address;
+use std::collections::{BTreeMap, HashMap};
+
+#[derive(Debug, Clone, Default)]
+pub struct AllocationIdTracker {
+    id_to_fee: HashMap<Address, u128>,
+    fee_to_count: BTreeMap<u128, u32>,
+    total_fee: u128,
+}
+
+impl AllocationIdTracker {
+    pub fn add_or_update(&mut self, id: Address, fee: u128) {
+        if let Some(&old_fee) = self.id_to_fee.get(&id) {
+            self.total_fee -= old_fee;
+            *self.fee_to_count.get_mut(&old_fee).unwrap() -= 1;
+            if self.fee_to_count[&old_fee] == 0 {
+                self.fee_to_count.remove(&old_fee);
+            }
+        }
+
+        if fee > 0 {
+            self.id_to_fee.insert(id, fee);
+            self.total_fee += fee;
```

Comment:
trigger value is currently a `u64`. Even if the trigger value is in MAX, this should trigger the rav request way earlier. 

the only possible attack vector is if the receipt contains the max value, which in this case this should have been rejected by the indexer-service because of not enough funds. (I'm not should if this is the case)

Reply:
`indexer-service` currently does not check the value at all, nor does it check the available funds (just checks that the escrow is > 1).
We should add a hard limit for receipt fee value on indexer-service though (user set value).

Reply:
Just added an overflow check and refactor the logic to be simpler and efficient only on the most used part (update)